### DISCUSSION TITLE
About MIXED element storage

### DISCUSSION THREAD

**Discussion by MicK7** [CGNS/CGNS#281] [CGNStalk]
Currently MIXED Elements use a "stream" like storage. Indeed, it interlaces the type of the element with the point connectivity.
As I understand, this strategy is meant to have fewer array stored (one connectivity + the offset for partial reading since CGNS 4.0) and perhaps be more compact. However for large number of elements switching to int64 for point indexing is necessary. As CGNS is currently design it forces also the switch on the element type thus the compacity argument is lost.
Recently lots of code (solver and postprocessor) have changed there historical implementation of connectivity to two separated array: one for point connectivity and the other for celltype (and an offset array for partial read)
Would there be an interest to do such a switch in CGNS and add support to a new MIXED Element section type ?

Current MIXED Element Node:
   ElementConnectivity [5, 1, 2, 3, 5, 4, 5, 6 ]
   ElementStartOffset [0,  4,  8]

New MIXEDS Element Node: (S stands for Separated)
   ElementConnectivity [1, 2, 3, 4, 5, 6 ]
   ElementTypes [5, 5,]
   ElementStartOffset [0,  3,  6]

In this case ElementTypes can remain as an int32 value even if you go for an exascale computation ....

**Comment by brtnfld** [comment]
I don't think it would hurt to have an option for a more compact data structure, reducing write times. I'm not sure of the current demand for it, though. I would think that using a partitioned mesh would possibly eliminate the need to use int64.
